server/update-workflow-interface/update-workflow-interface.md

```diff
@@ -0,0 +1,217 @@
+# Proposal: Update Workflow Server Interface
+
+Currently (as of version 1.15), ad-hoc runtime interaction with Temporal
+workflows is achieved through queries and signals. The former technique can
+interrogate the state of a workflow and returns a result but must not _change_
+the state of the workflow, while the latter can change workflow state but does
+not natively support return values [^1]. 
+
+The missing piece in runtime workflow interaction is an affordance that
+provides both (1) the ability to change workflow state, and (2) a mechanism to
+observe the result of the same without polling or setting up callback
+structures. This feature will be workflow updates.
+
+This proposal describes the server API (i.e. gRPC) for such a feature. We leave
+implementation details for a deeper design review and instead relate the API,
+its expected usage and the guarantees provided.
+
+## Concepts
+
+### Update
+
+An Update is the execution of arbitrary workflow logic (subject to the standard
+workflow code constraints on determinism) triggered by an UpdateRequest and
+eventually resulting in a single UpdateResponse. The processing of an Update is
+performed by workflow code and as such the timeliness of the UpdateResponse is
+dependent on worker resources being available to process the Update. A given
+Update applies to exactly one workflow execution. The lifetime of an updated
+begins at the first UpdateRequest triggering said Update and ends when the
+UpdateResponse has been mad durable. Consequently, the lifetime of an Update may
```

Comment:
_mad_ durable, yo 😎 

Lol forgive me, was browsing through this on my phone and just had to get this one in before you fix it.

Reply:
Do not sleep on this level of durability. I kind of want to keep it.